pynest/examples/sensitivity_to_perturbation.py

```diff
@@ -0,0 +1,231 @@
+time period 'fade_out'. In order to suppress the firing after the
+simulation successfully the fade out period has to last at least two
+times the simulation resolution.
+'''
+
+ext = nest.Create("poisson_generator",
+                  params={'rate':rate_ext, 'stop': T})
+nest.Connect(ext, allnodes,
+                  syn_spec={'weight':Jext, 'delay':dt})
+
+suppr = nest.Create("dc_generator",
+                  params={'amplitude':-1e16, 'start': T, 'stop':T+fade_out})
+nest.Connect(suppr, list(allnodes))
+
+spikedetector = nest.Create("spike_detector")   
+nest.ConvergentConnect(allnodes, spikedetector)
+
+'''
+Creating the spike generator that provides the extra spike (perturbation).
+'''
+stimulus = nest.Create("spike_generator")
+nest.SetStatus(stimulus, {'spike_times': []})
+
+'''
+Finally we run the two simulations successively. After each simulation the
+sender ids and spiketimes are stored in a list ('senders', 'spiketimes').
+'''
+
+senders = []
+spiketimes = []
+
+for trial in [0,1]:
+    '''
+    As mentioned above, we need to reset the network, the random number
+    generator, and the clock of the Kernel. In addition we make sure 
+    that there is no spike left in the spike generator.
+    '''
+    nest.ResetNetwork()
+    nest.SetStatus([0], [{"rng_seeds": [seed_NEST]}])
+    nest.SetStatus([0], {'time': 0.0})
+    nest.SetStatus(spikedetector, {'n_events':0})
+
+    '''
+    We assign random initial membrane potentials to all neurons (uniform
+    disbributed between Vmin and Vthreshold)
+    '''
+
+    numpy.random.seed(seed_numpy)
+    Vms = Vmin + (Vmax - Vmin) * numpy.random.rand(N)
+    nest.SetStatus(allnodes, "V_m", Vms)
+
+    '''
+    In the second trial we add an extra input spike at time t_stim to the neuron
+    that fires first after perturbation time t_stim. Thus, we make sure that the
+    perturbation is transmitted to the network before it fades away in the 
+    perturbed neuron. (Single IAF-neurons are not chaotic.)
+    '''
+
+    if trial == 1:
+        id_stim = [senders[0][spiketimes[0] > t_stim][0]]
+        nest.Connect(stimulus, list(id_stim), syn_spec={'weight':Jstim, 'delay':dt})
+        nest.SetStatus(stimulus, {'spike_times':[t_stim]})
+
+    '''
+    Now we simulate the network and add a fade out period to get rid of the 
+    remaining spikes in the current NEST memory (see introduction).
+    '''
+
+    nest.Simulate(T)
+    nest.Simulate(fade_out)  
+
+    '''
+    Storing the data.
+    '''
+
+    senders += [nest.GetStatus(spikedetector, 'events')[0]['senders']]
+    spiketimes += [nest.GetStatus(spikedetector, 'events')[0]['times']]
+
+'''
+Plotting the spiking activity of the network (first trial in red, second trial in black).
+'''
+
+pylab.figure(1)
+pylab.clf()
+pylab.plot(spiketimes[0], senders[0], 'ro', ms=4.)
+pylab.plot(spiketimes[1], senders[1], 'ko', ms=2.)
+pylab.xlabel('time (ms)')
+pylab.ylabel('neuron id')
+pylab.xlim((0, T))
+pylab.ylim((0, N))
+pylab.show()
```

Comment:
Please remove the call to show(). This will make the CI choke, as there's no display attached there.